mm2src/coins/z_coin.rs

```diff
@@ -942,11 +979,21 @@ impl<'a> UtxoCoinBuilder for ZCoinBuilder<'a> {
         );
 
         let blocks_db = self.init_blocks_db().await?;
+        let locked_notes_db_path = self.db_dir_path.join(format!("{}_locked_notes_cache.db", self.ticker));
```

Comment:
Depending on which PR get merged first, this one or https://github.com/KomodoPlatform/komodo-defi-framework/pull/2398, we will need some changes in either. c.c. @mariocynicys

Reply:
This is not needed for wasm @borngraced

Reply:
Not address dir? These are locked notes related to an address.

Reply:
better to have it in wallet_dir since this is an extension of the functionality covered by `walled.db` which is managed by libzcash and not us. this `wallet.db` works on accounts/HD level and not just address level, so its extension DB should also work on accounts level.

p.s. `wallet.db` has tables called `sent_notes` and `received_notes`, which i guess complements the data found in `locked_notes` introduced in this PR.

Reply:
I understand, I guess the `rseed` will be unique even for multiple addresses so this approach should work fine.

Reply:
double checked in the code. it's generated at random.
also specified formally here in page 29 https://zips.z.cash/protocol/protocol.pdf